### PR TITLE
add addon recommendation endpoint

### LINKED ISSUE #8118: Implement recommendations API with A/B
We need an API that returns add-on recommendations from a single guid. We want to do A/B testing on this so it also needs some way for the frontend to ask for a fallback list of add-ons instead, managed by addons-server, which is the "B" side of things.

Rough implementation plan:
- inherit from `AddonSearchView`
- implement different filter_backends: keep `ReviewedContentFilter` but add a custom filter that calls TAAR if necessary (or just use the fallback guids if we're on the "B" side of the A/B test - or if TAAR took too long to answer) and then just search for the guids returned
- implement a custom serializer, if only to add a property indicating whether we were using the fallback or not.

### PR DESCRIPTION
**Pull request by eviljeff**
Fixes #8118

### CODE REVIEW

File: docs/topics/api/addons.rst [hunk 1]
```diff
@@ -527,3 +527,24 @@ Compatibilty overrides are used within Firefox i(and other toolkit applications
...
+
+This endpoint provides recommendations of other addons to install, fetched from the `recommendation service <https://github.com/mozilla/taar>`_.
+Four recommendations are fetched, but only valid, publicly available addons are shown (so max 4 will be returned, and possibly less).
+
+.. http:get:: /api/v3/addons/recommendations/
+
+    :query string guid: Fetch recommendations for this add-on guid.
+    :query string lang: Activate translations in the specific language for that query. (See :ref:`translated fields <api-overview-translations>`)
+    :query string variant: This endpoint is part of an A/B test.  `a` is fetch recommendations from the recommendation service; `b` is return a fixed fallback.
+    :>json string a_b_outcome: The A/B test outcome.  Will be either: `taar` - responses from recommendation service; `timeout` - service timed out so returned fallback; `fallback` - `b` variant was requested so fallback returned.
```
Comment (bobsilverberg) [inline docs/topics/api/addons.rst]: As per our discussion, the three possible values for outcome will be `recommended`, `recommended_fallback` and `curated`.

File: docs/topics/api/addons.rst [hunk 2]
```diff
@@ -527,3 +527,24 @@ Compatibilty overrides are used within Firefox i(and other toolkit applications
...
+.. _addon-recommendations:
+
+This endpoint provides recommendations of other addons to install, fetched from the `recommendation service <https://github.com/mozilla/taar>`_.
+Four recommendations are fetched, but only valid, publicly available addons are shown (so max 4 will be returned, and possibly less).
+
+.. http:get:: /api/v3/addons/recommendations/
+
+    :query string guid: Fetch recommendations for this add-on guid.
+    :query string lang: Activate translations in the specific language for that query. (See :ref:`translated fields <api-overview-translations>`)
+    :query string variant: This endpoint is part of an A/B test.  `a` is fetch recommendations from the recommendation service; `b` is return a fixed fallback.
```
Comment (bobsilverberg) [inline docs/topics/api/addons.rst]: As per our discussion, the variable name should be `recommended` with a value of `true` or `false`.

File: docs/topics/api/addons.rst (hunk 1 above)
Comment (bobsilverberg) [inline docs/topics/api/addons.rst]: We should also return a `fallback_reason`, which will be one of `timeout` or `no_results`.

Comment (eviljeff) [reply docs/topics/api/addons.rst]: https://github.com/mozilla/addons-server/pull/8126/commits/06ed8d22dc8ac7c30b4901beaaeb2abaedb86b62

File: docs/topics/api/addons.rst (hunk 2 above)
Comment (eviljeff) [reply docs/topics/api/addons.rst]: https://github.com/mozilla/addons-server/pull/8126/commits/06ed8d22dc8ac7c30b4901beaaeb2abaedb86b62

File: docs/topics/api/addons.rst (hunk 1 above)
Comment (eviljeff) [reply docs/topics/api/addons.rst]: https://github.com/mozilla/addons-server/pull/8126/commits/3da83bc15025a23f6995afc4a64a983913316009

**Comment by eviljeff** [comment]
@diox https://github.com/mozilla/addons-server/pull/8126/commits/34071a98a712a5ec1427c5e8ca105184ac5740f0

**Pull request merged**